packages/ckeditor5-dev-release-tools/tests/release/updateversions.js

```diff
@@ -0,0 +1,138 @@
+				'/ckeditor5-dev'
+			] );
+
+			updateVersions( { version: '1.0.1', packagesDirectory: 'packages' } );
+
+			expect( syncStub.firstCall.args[ 0 ] ).to.deep.equal( [ 'package.json', 'packages/*/package.json' ] );
+			expect( outputJsonSyncStub.callCount ).to.equal( 4 );
+			expect( outputJsonSyncStub.firstCall.args[ 0 ] ).to.contain( 'package1' );
+			expect( outputJsonSyncStub.firstCall.args[ 1 ] ).to.deep.equal( { version: '1.0.1' } );
+		} );
+
+		it( 'should call outputJsonSyncStub 1 time when the version is correct packages are undefined and 1 root', () => {
+			syncStub.returns( [ '/ckeditor5-dev' ] );
+
+			updateVersions( { version: '1.0.1' } );
+
+			expect( syncStub.firstCall.args[ 0 ] ).to.deep.equal( [ 'package.json' ] );
+			expect( outputJsonSyncStub.callCount ).to.equal( 1 );
+			expect( outputJsonSyncStub.firstCall.args[ 0 ] ).to.contain( 'ckeditor5-dev' );
+			expect( outputJsonSyncStub.firstCall.args[ 1 ] ).to.deep.equal( { version: '1.0.1' } );
+		} );
+
+		it( 'should throw when npm show returns that version is already in use', () => {
+			getPackageJsonStub.returns( { version: '1.0.0', name: 'stub-package' } );
+			shExecStub.returns( '' );
+
+			expect( () => updateVersions( { version: '1.0.1' } ) )
+				.to.throw( Error, 'Provided version 1.0.1 is already used in npm by stub-package' );
+		} );
+
+		it( 'should not throw when npm show returns that version is not in use', () => {
+			shExecStub.throws( new Error( 'is not in this registry' ) );
+			getPackageJsonStub.returns( { version: '1.0.0', name: 'stub-package' } );
+
+			expect( () => updateVersions( { version: '1.0.1' } ) ).to.not.throw( Error );
+		} );
+
+		it( 'should throw when npm show throws another error', () => {
+			shExecStub.throws( new Error( 'custom error' ) );
+			getPackageJsonStub.returns( { version: '1.0.0', name: 'stub-package' } );
+
+			expect( () => updateVersions( { version: '1.0.1' } ) ).to.throw( Error );
+		} );
+
+		it( 'should call outputJsonSyncStub when version is 0.0.0-', () => {
+			updateVersions( { version: '0.0.0-nightly-123.0' } );
+
+			expect( outputJsonSyncStub ).to.be.called;
+		} );
+
+		it( 'should throw when new version is not greater than old version', () => {
+			gtStub.returns( false );
+
+			expect( () => updateVersions( { version: '1.0.0' } ) )
+				.to.throw( Error, 'Provided version 1.0.0 must be greater than 1.0.0.' );
+		} );
+
+		it( 'should throw when new version is not a valid semver version', () => {
+			validStub.returns( false );
+
+			expect( () => updateVersions( { version: '1.0.0' } ) )
+				.to.throw( Error, 'Provided version 1.0.0 must be a valid semver version.' );
+		} );
+
+		it( 'should pass cwd to globSync', () => {
+			updateVersions( { version: '1.0.1', cwd: 'Users/username/ckeditor5-dev/custom-dir' } );
+
+			expect( syncStub.firstCall.args[ 1 ] ).to.deep.equal( {
+				cwd: 'Users/username/ckeditor5-dev/custom-dir',
+				absolute: true,
+				nodir: true
+			} );
```

Comment:
This assertion should be duplicated in the first two tests. Then, this test scenario can be removed.

Reply:
I accepted almost all of the comments, but this will be an exception :)
Adding this test logic to test 1 and 2 would result in:
1. Duplicating test code.
2. Cluttering the test with unnecessary logic.

Adding a new test just for cwd looks cleaner :)